macos/mac_os_resource_snd.ksy

```diff
@@ -0,0 +1,445 @@
+        value: 0x07
+        doc: wave table only, Sound Manager 2.0 and earlier
+      wave_init:
+        value: options & wave_init_channel_mask
+        enum: wave_init_option
+        if: id == data_type::wave_table_synth
+      init_pan_mask:
+        -orig-id: initPanMask
+        value: 0x0003
+        doc: mask for right/left pan values
+      pan_init:
+        value: options & init_pan_mask
+        enum: init_option
+      init_stereo_mask:
+        -orig-id: initStereoMask
+        value: 0x00C0
+        doc: mask for mono/stereo values
+      stereo_init:
+        value: options & init_stereo_mask
+        enum: init_option
+      init_comp_mask:
+        -orig-id: initCompMask
+        value: 0xFF00
+        doc: mask for compression IDs
+      comp_init:
+        value: options & init_comp_mask
+        enum: init_option
+  sound_command:
+    seq:
+      - id: is_data_offset
+        type: b1
+      - id: cmd
+        type: b15
+        enum: cmd_type
+      - id: param1
+        type: u2
+      - id: param2
+        type: u4
+    instances:
+      sound_header:
+        pos: param2
+        type: sound_header
+        size-eos: true
+        if: is_data_offset and cmd == cmd_type::buffer_cmd
+  sound_header:
+    seq:
+      - id: sample_ptr
+        -orig-id: samplePtr
+        type: u4
+        doc: pointer to samples (or 0 if samples follow data structure)
+      - id: num_samples
+        -orig-id: length
+        type: u4
+        doc: number of samples
+        if: sound_header_type == sound_header_type::standard
+      - id: num_channels
+        -orig-id: numChannels
+        type: u4
+        doc: number of channels in sample
+        if: sound_header_type == sound_header_type::extended or sound_header_type == sound_header_type::compressed
+      - id: sample_rate
+        -orig-id: sampleRate
+        type: unsigned_fixed_point
+        doc: The rate at which the sample was originally recorded.
+      - id: loop_start
+        -orig-id: loopStart
+        type: u4
+        doc: loop point beginning
+      - id: loop_end
+        -orig-id: loopEnd
+        type: u4
+        doc: loop point ending
+      - id: encode
+        type: u1
+        enum: sound_header_type
+        doc: sample's encoding option
+      - id: midi_note
+        type: u1
+        doc: base frequency of sample, expressed as MIDI note values, 60 is middle C
+        -orig-id: baseFrequency
+      - id: extended_or_compressed
+        type: extended_or_compressed
+        if: sound_header_type == sound_header_type::extended or sound_header_type == sound_header_type::compressed
+      - id: sample_area
+        -orig-id: sampleArea
+        size: |
+          sound_header_type == sound_header_type::standard ? num_samples :
+          sound_header_type == sound_header_type::extended ? extended_or_compressed.num_frames * num_channels * extended_or_compressed.bits_per_sample / 8 :
+          _io.size - _io.pos
+        doc: sampled-sound data
```

Comment:
`sample_area` is present here only if `sample_ptr == 0`, right?

```suggestion
        if: sample_ptr == 0
        doc: sampled-sound data
```

Also please create an `instance` with `pos: sample_ptr` parsing the samples area if `sample_ptr != 0`.

Reply:
I think samplePtr points to a memory location, not a file location, so I don't think I can just take the value in an instance pos. Additionally in the case of a compressed header, I don't know the compressed size:

> A pointer to the sampled-sound data. If the sampled sound is located in memory immediately after the futureUse4 field, then this field should be set to NIL. Otherwise, this field is a pointer to the memory location of the sampled-sound data.

https://developer.apple.com/library/archive/documentation/mac/pdf/Sound/Sound_Manager.pdf 2-107

Reply:
@tisoft 

> I think samplePtr points to a memory location, not a file location, so I don't think I can just take the value in an instance pos. Additionally in the case of a compressed header, I don't know the compressed size

OK, fine, the Mac OS sound resource is really a weird format 😃 Let it be then.